### PR TITLE
Lowering `aten.stack.default` op

### PR DESCRIPTION
**Pull request by pschang-phy**
### Ticket
https://github.com/tenstorrent/pytorch2.0_ttnn/issues/596

### Problem description
`aten.stack.default` not lowering yet.

### What's changed
`aten.stack.default` op is transform to equivalent computation
1. reshape all tensors with specified dimension
2. concat all unsqueezed tensors

### CODE REVIEW

File: tests/lowering/tensor_manipulation/test_stack.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    def forward(self, tensors, dim):
+        return torch.stack(tensors, dim)
+
+
+@pytest.mark.parametrize(
+    "input_shape, dim",
+    [
+        ((12, 16), -1),
+    ],
```
Comment (ayerofieiev-tt) [inline tests/lowering/tensor_manipulation/test_stack.py:19]: can we get a couple more corner cases captured here?

**Review by ayerofieiev-tt — APPROVED** [review]

File: tests/lowering/tensor_manipulation/test_stack.py (hunk 1 above)
Comment (pschang-phy) [reply tests/lowering/tensor_manipulation/test_stack.py:19]: ok, I will add more different cases
